lib/spreewald/web_steps.rb

```diff
@@ -217,14 +217,23 @@
 #
 # See [here](https://makandracards.com/makandra/1225-test-that-a-number-or-money-amount-is-shown-with-cucumber) for details
 Then /^I should( not)? see the (?:number|amount) ([\-\d,\.]+)(?: (.*?))?$/ do |negate, amount, unit|
-  no_minus = amount.starts_with?('-') ? '' : '[^\\-]'
-  nbsp = " "
-  regexp = Regexp.new(no_minus + "\\b" + Regexp.quote(amount) + (unit ? "( |#{nbsp}|&nbsp;)(#{unit}|#{Regexp.quote(HTMLEntities.new.encode(unit, :named))})" :"\\b"))
-  expectation = negate ? :not_to : :to
+  expect_to_match = negate ? :not_to : :to
+  amount_included_minus = amount.gsub!(/^-/, '')
+  expect_minus = amount_included_minus ? '-' : '(?:[^\\-]|^)'
+  word_boundary = '\b'
+  expect_unit_or_boundary = if unit
+    nbsp = " "
+    "( |#{nbsp})(#{unit})"
+  else
+    word_boundary
```

Comment:
It's just a minor thing but I don't see the value of using a variable here. I guess it is a leftover of changing things. I would prefer to have the `'\b' ` directly in the else clause.